sklearn/neighbors/tests/test_lof.py

```diff
@@ -258,6 +258,26 @@ def test_sparse():
     lof.fit_predict(X)
 
 
+def test_lof_error():
+    X = [[0, 1, 2, 3]]
+    clf = neighbors.LocalOutlierFactor(contamination="auto", n_neighbors=2)
```

Comment:
This test is confusing

First, we have `n_neighbors=2` but the error message says `n_neighbors=1`.
This is because `LocalOutlierFactor.fit` silently uses `self.n_neighbors = max(1, min(n_neighbors, n_samples - 1))`. We should be more explicit in the test:
```py
X = np.ones((7, 7))
# If `n_neighbors == n_samples > 1`, the fit does not raise an error.
neighbors.LocalOutlierFactor(n_neighbors=2).fit(X[:2])
# If `n_neighbors == n_samples == 1`, the fit does raise an error.
with pytest.raises(ValueError, match=msg.format(inequality_str)):
    neighbors.LocalOutlierFactor(n_neighbors=1).fit(X[:1])
```